### PR TITLE
Handle the nutanix provider case in cluster reconciler

### PR DESCRIPTION
**Pull request by thunderboltsid**
Currently, this is more of a placeholder to start with as the TemplateResources method
just returns an empty array.

### CODE REVIEW

**Comment by thunderboltsid** [comment]
@abhinavmpandey08 the test are passing, anything else we need here?

File: controllers/resource/reconciler.go [hunk 1]
```diff
@@ -146,6 +151,37 @@ func (cor *clusterReconciler) Reconcile(ctx context.Context, objectKey types.Nam
 			return err
 		}
 		resources = append(resources, r...)
+	case anywherev1.NutanixDatacenterKind:
```
Comment (abhinavmpandey08) [inline controllers/resource/reconciler.go:154]: Can you add a feature gate here similar to this? https://github.com/aws/eks-anywhere/blob/main/controllers/resource/reconciler.go#L109-L111

**Review by abhinavmpandey08 — APPROVED** [review]
/approve

**Pull request merged**